pkg/execute/executor.go

```diff
@@ -267,19 +193,18 @@ func (e *DefaultExecutor) Execute(ctx context.Context) interactive.Message {
 		cmdRes = strings.ToLower(cleanArgs[1])
 	}
 
-	resources, found := e.cmdsMapping[cmdVerb]
-	if !found {
+	fn, foundRes, foundFn := e.cmdsMapping.FindFn(cmdVerb, cmdRes)
+	if !foundRes {
 		e.reportCommand(anonymizedInvalidVerb, false)
 		e.log.Infof("received unsupported command: %q", execFilter.FilteredCommand())
 		return respond(unsupportedCmdMsg, cmdCtx)
 	}
 
-	fn, found := resources[cmdRes]
-	if !found {
+	if !foundFn {
 		e.reportCommand(fmt.Sprintf("%s {invalid resource}", cmdVerb), false)
 		e.log.Infof("received unsupported resource: %q", execFilter.FilteredCommand())
 		msg := incompleteCmdMsg
-		if helpMessage, ok := availableHelpMessages[cmdVerb]; ok {
+		if helpMessage := e.cmdsMapping.HelpMessageForVerb(cmdVerb, botName); helpMessage != "" {
 			msg = helpMessage
 		}
```

Comment:
maybe we can move the `incompleteCmdMsg` to `HelpMessageForVerb` and return it in case there is no other help message. As a result, we make this flow more straight.
```go
msg := e.cmdsMapping.HelpMessageForVerb(cmdVerb, botName) // it's up to the help provider to resolve the final message
```